Source/JavaScriptCore/wasm/WasmIPIntGenerator.h

```diff
@@ -247,6 +260,77 @@ enum class UIntBytecode: uint8_t {
     NumOpcodes // this must be the last element of the enum!
 };
 
+// GC Metadata
+
+struct StructNewMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint16_t params;
+    uint8_t length;
+};
+
+struct StructNewDefaultMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint8_t length;
+};
+
+struct StructGetSetMetadata {
+    uint32_t fieldIndex;
+    uint8_t length;
+};
+
+struct ArrayNewMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint8_t length;
+};
+
+struct ArrayNewFixedMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint32_t arraySize;
+    uint8_t length;
+};
+
+struct ArrayNewDataMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint32_t dataSegmentIndex;
+    uint8_t length;
+};
+
+struct ArrayNewElemMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint32_t elemSegmentIndex;
+    uint8_t length;
+};
+
+struct ArrayGetSetMetadata {
+    Wasm::TypeIndex typeIndex;
+    uint8_t length;
+};
+
+struct ArrayFillMetadata {
+    uint8_t length;
+};
+
+struct ArrayCopyMetadata {
+    uint8_t length;
+};
+
+struct ArrayInitDataMetadata {
+    uint32_t dataSegmentIndex;
+    uint8_t length;
+};
+
+struct ArrayInitElemMetadata {
+    uint32_t elemSegmentIndex;
+    uint8_t length;
+};
+
+struct RefTestCastMetadata {
+    int32_t typeIndex;
+    bool allowNull;
```

Comment:
This isn't needed by the asm code, right? Can the cpp helper just look this up from the wasm type index? Alternatively, can we arrange these structs to be more compact?

Reply:
Yup! Turns out, our `shouldNegate` flag is solely just for LLInt to handle `br_on_cast`/`br_on_cast_fail` without adding a custom instruction for those... I'm working on the patches now but now we should only need a type index and length. Thanks for the catch :)